src/app/services/task.service.spec.ts

```diff
@@ -0,0 +1,149 @@
+// @angular
+import { TestBed } from '@angular/core/testing';
+
+// services
+import { TaskService } from './task.service';
+import { WebService } from './web.service';
+
+// models
+import { Task } from '../models/task.model';
+
+// mocks
+import { WebServiceMock } from '../mocks/mock.web.service';
+
+let taskService: TaskService;
+let webService: WebService;
+
+const task: Task = { name: 'task 1', isCompleted: true};
+const tasks: Task[] = [
+  { name: 'task 1', isCompleted: true },
+  { name: 'task 2', isCompleted: false },
+  { name: 'task 3', isCompleted: false },
+  { name: 'task 4', isCompleted: true },
+];
+
+describe('TaskService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        TaskService,
+        { provide: WebService, useClass: WebServiceMock }
+      ]
+    });
+    taskService = TestBed.get(TaskService);
+    webService = TestBed.get(WebService);
+    taskService.tasks = tasks;
+  });
+
+  it('should be created', () => {
+    expect(taskService).toBeTruthy();
+  });
+
+  it('call renderAllTasks() => do not call WebService.getTasks() if local storage return array of tasks', () => {
+    spyOn(localStorage, 'getItem').and.returnValue(JSON.stringify(tasks));
+    const spy = spyOn(webService, 'getTasks').and.callThrough();
+    taskService.renderAllTasks();
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  it('call renderAllTasks() => call WebService.getTasks() if local storage return empty array', () => {
+    spyOn(localStorage, 'getItem').and.returnValue('[]');
+    const spy = spyOn(webService, 'getTasks').and.callThrough();
+    taskService.renderAllTasks();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('call renderAllTasks() => call WebService.getTasks() if local storage return null', () => {
+    spyOn(localStorage, 'getItem').and.returnValue(null);
+    const spy = spyOn(webService, 'getTasks').and.callThrough();
+    taskService.renderAllTasks();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('call addTask() => call TaskService.updateLocalStorage()', () => {
+    const spy = spyOn(taskService, 'updateLocalStorage').and.stub();
+    taskService.addTask(task);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('call addTask() => add addedTask to tasks', () => {
+    const addedTask = {name: 'rrrr', isCompleted: false};
+    taskService.addTask(addedTask);
+    expect(taskService.tasks).toContain(addedTask);
+  });
+
+  it('call toggleTask() => call TaskService.updateLocalStorage()', () => {
+    const spy = spyOn(taskService, 'updateLocalStorage').and.stub();
+    taskService.toggleTask(task);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('call toggleTask() => toggle task.isCompleted', () => {
+    const status = task.isCompleted;
+    taskService.toggleTask(task);
+    expect(status).toBe(!task.isCompleted);
```

Comment:
You should compare to false
`task.isCompleted = true;
 taskService.toggleTask(task);
expect(task.isCompleted).toBe(falsy);`

Reply:
Nope

Reply:
rewrite. add checking   true -> toggle -> false